src/ray/util/signal_handler.h

```diff
@@ -0,0 +1,53 @@
+#ifndef RAY_UTIL_SIGNAL_HANDLER_H
+#define RAY_UTIL_SIGNAL_HANDLER_H
+
+#include <signal.h>
+#include <string>
+#include <vector>
+
+namespace ray {
+
+class SignalHandlers {
+ public:
+  /// Setup the signal handler, which should be called in main function.
+  ///
+  /// \param app_name The app name that install the signal handler.
+  /// \param install_sigterm Whether install the handler for SIGTERM, because
+  /// some app has already have a handler for this signal.
+  /// \return Void.
+  static void InstallSingalHandler(const std::string &app_name,
```

Comment:
If we install/unistall signal handlers by RAII, let's declared `InstallSingalHandler()` and `UninstallSingalHandler()` as private member functions according `Minimum interfaces exposure principle`.

The users of class `SignalHandlers` shouldn't know `InstallSingalHandler()` and `UninstallSingalHandler()`.